pkl-cli/pkl-cli.gradle.kts

```diff
@@ -160,47 +160,29 @@ fun Exec.configureExecutable(
         "--initialize-at-build-time="
         // needed for messagepack-java (see https://github.com/msgpack/msgpack-java/issues/600)
-        ,"--add-opens=java.base/java.nio=ALL-UNNAMED"
-        ,"--add-opens=java.base/sun.nio.ch=ALL-UNNAMED"
+        ,"--initialize-at-run-time=org.msgpack.core.buffer.DirectBufferAccess"
         ,"--no-fallback"
         ,"-H:IncludeResources=org/pkl/core/stdlib/.*\\.pkl"
         ,"-H:IncludeResources=org/jline/utils/.*"
         ,"-H:IncludeResources=org/pkl/certs/PklCARoots.pem"
-        //,"-H:IncludeResources=org/pkl/core/Release.properties"
         ,"-H:IncludeResourceBundles=org.pkl.core.errorMessages"
         ,"--macro:truffle"
         ,"-H:Class=org.pkl.cli.Main"
         ,"-H:Name=${outputFile.get().asFile.name}"
-        //,"--native-image-info"
-        //,"-Dpolyglot.image-build-time.PreinitializeContexts=pkl"
         // the actual limit (currently) used by native-image is this number + 1400 (idea is to compensate for Truffle's own nodes)
         ,"-H:MaxRuntimeCompileMethods=1800"
         ,"-H:+EnforceMaxRuntimeCompileMethods"
         ,"--enable-url-protocols=http,https"
-        //,"--install-exit-handlers"
         ,"-H:+ReportExceptionStackTraces"
-        ,"-H:-ParseRuntimeOptions" // disable automatic support for JVM CLI options (puts our main class in full control of argument parsing)
-        //,"-H:+PrintAnalysisCallTree"
-        //,"-H:PrintAnalysisCallTreeType=CSV"
-        //,"-H:+PrintImageObjectTree"
-        //,"--features=org.pkl.cli.svm.InitFeature"
-        //,"-H:Dump=:2"
-        //,"-H:MethodFilter=ModuleCache.getOrLoad*,VmLanguage.loadModule"
-        //,"-g"
-        //,"-verbose"
-        //,"--debug-attach"
-        //,"-H:+AllowVMInspection"
-        //,"-H:+PrintHeapHistogram"
-        //,"-H:+ReportDeletedElementsAtRuntime"
-        //,"-H:+PrintMethodHistogram"
-        //,"-H:+PrintRuntimeCompileMethods"
-        //,"-H:NumberOfThreads=1"
-        //,"-J-Dtruffle.TruffleRuntime=com.oracle.truffle.api.impl.DefaultTruffleRuntime"
-        //,"-J-Dcom.oracle.truffle.aot=true"
-        //,"-J:-ea"
-        //,"-J:-esa"
-        // for use with https://www.graalvm.org/docs/tools/dashboard/
-        //,"-H:DashboardDump=dashboard.dump", "-H:+DashboardAll"
+        // disable automatic support for JVM CLI options (puts our main class in full control of argument parsing)
+        ,"-H:-ParseRuntimeOptions"
+        // quick build mode: 40% faster compilation, 20% smaller (but presumably also slower) executable
+        ,"-Ob"
+        // Use Intel Skylake (2015) CPU features, which should also be supported by AMD ZEN (2017) CPUs.
+        // Tested on ZEN 2.
+        // Feature list: CX8, CMOV, FXSR, MMX, SSE, SSE2, SSE3, SSSE3, SSE4_1, SSE4_2, POPCNT,
+        // LZCNT, AVX, AVX2, AES, CLMUL, BMI1, BMI2, FMA, AMD_3DNOW_PREFETCH, ADX, FLUSHOPT
+        , if (buildInfo.arch == "amd64") "-march=skylake" else ""
```

Comment:
What does this do? Does this mean native executables are not compatible with Intel CPUs older than 2015, and AMD CPUs older than 2017?

Reply:
Yes, this raises the baseline for supported CPUs from "x64-v3" (default) to "skylake". In return, the native image compiler can use many additional CPU features/instructions, resulting in faster and more efficient assembly code. 
I arrived at this by checking the output of `-march=list`. "skylake" seemed to be a good middle ground between supporting a wide range of CPUs and leveraging many CPU features.

https://www.graalvm.org/latest/reference-manual/native-image/overview/BuildOutput/#recommendation-cpu

Reply:
These CLIs are libraries, and have a broad set of use-cases. For that reason, we should probably be conservative here. I'm thinking we should hold off on this change.

CC @holzensp @stackoverflow for comments

Reply:
Agree with Dan that we should be conservative here, even though skylake is almost 9 years old. Specially without any performance data to compare. If the performance bump is noticeable we can create a new architecture release or just drop old CPUs in the new version.

Reply:
I agree with @bioball and @stackoverflow. We're also considering spending a version-cycle on GraalVM/Truffle/native-image updates and perf (including binary size). For now, let's be quite conservative.